src/logger-mixin/mixins/logger-mixin.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// tslint:disable:no-any
+
+import {Constructor} from '@loopback/context';
+
+/**
+* A mixin class for Application that creates a .logger()
+* function to register a Logger automatically. Also overrides
+* component function to allow it to register Logger's automatically.
+*
+* ```ts
+*
+* class MyApplication extends LoggerMixin(Application) {}
+* ```
+*/
+export function LoggerMixin<T extends Constructor<any>>(superClass: T) {
```

Comment:
If this is meant to be used with Applications (or implementations of Application), then it should be of type `Constructor<Application>`.

Reply:
+1 for using a more specific type than `any`.

I don't think we can use `Application` here, because we may be referring to a different constructor class (in our copy of `@loopback/core` in `node_modules` tree) than the constructor class used by the application using our mixin (from their copy of `@loopback/core` in their `node_modules` tree).

Considering that our own `@loopback/repository` is using `T extends Constructor<any>` ([source](https://github.com/strongloop/loopback-next/blob/fa3026de83991e4de518ae70554ed2a56835fa3d/packages/repository/src/repository-mixin.ts#L21)), I think it's ok to do the same here, at least for now.

Reply:
Using type Application throws a whole series of errors such as using internal type `ApplicationOptions`, etc. I think it's because we don't have an interface for `Application`. Even setting it to `Constructor<{}> results in errors.